src/lib/include/template_func.tpp

```diff
@@ -0,0 +1,19 @@
+#ifndef XDIST_DISTOPIA_TEMPLATE_H
+#define XDIST_DISTOPIA_TEMPLATE_H
+
+template <typename T>
+float SingleTemplatePairwiseDistance(const T *coords1, const T *coords2,
+                                     const T *box) {
+  T dx = 0.0;
+
+  for (unsigned char i = 0; i < 3; ++i) {
+    T rij = coords1[i] - coords2[i];
+    T adj = round(rij / box[i]);
+    rij -= adj * box[i];
+    dx += rij * rij;
+  }
+  // sqrt must be defined for T
```

Comment:
needs to have a definition of sqrt for T or else you need to specialise the template for the T